src/Mod/Sketcher/App/planegcs/Geo.cpp

```diff
@@ -713,4 +712,64 @@ BSpline* BSpline::Copy()
     return crv;
 }
 
+double BSpline::getLinCombFactor(double x, size_t k, size_t i)
+{
+    // Adapted to C++ from the python implementation in the Wikipedia page for de Boor algorithm
+    // https://en.wikipedia.org/wiki/De_Boor%27s_algorithm.
+
+    // FIXME: This should probably be guaranteed by now, and done somewhere else
```

Comment:
> I am not against keeping it here, if there is a need to ensure the knots are flattened before calling the function.

I think the expression "flattened knots are obtained" is more appropriate, since `knots` and `mults` do not change. An assert might be best, but in the current state the code does rely on it being done here. Alternative places might be at the time `BSpline::knots` are defined.

Reply:
Side note: these are `VEC_D` instead of `VEC_pD` because for periodic knots we need to offset first/last few knots by the period and append them to the end/beginning. In the future, when the parametric value of knots are also changeable by solver (like weights are now), this will need to be changed as well, either making flattened knots every time, or making some data structures that account for the offset.

Reply:
OK. I understand it should be guaranteed in the future, pending future development, still needs to be addressed here today. There is an alternative place today, which may be better, but future development will render this also obsolete.

The maximum I would go today is to better reflect in the comment this exchange.

Reply:
I do not understand.

You indicate you have changed the FIXME to reflect this discussion. But I see the FIXME there. Is it a problem of not having pushed a commit?

I am happy with almost anything that does not say "FIXME", but rather indicates what you say above, that currently could be put in alternative locations, but that foreseen future development will require adaptation, and that currently it is still necessary somewhere, so we keep it here.

Reply:
What I find misleading is that it is indicated as "FIXME" implying that there is fixing to be done, when it is rather necessary that is there.

Reply:
I see. For now, I believe this can be changed to a TODO if you're more comfortable with that.

Coming to think of it, the ideal solution is to update `flattennedknots` every time knots or mult is changed. This could be done using getters/setters. This would remove setup knots from here. However that would require investigating wherever knots are accessed or modified.

Reply:
We can leave the FIXME with what you have introduced below. Thanks

Reply:
OK. Resolving this conversation then.